integration-tests/features/steps/component_analysis_v2.py

```diff
@@ -0,0 +1,163 @@
+    """Find the vulnerabilities report."""
+    json_data = context.response.json()
+
+    if "component_analyses" in json_data:
+        vulnerabilities = json_data['component_analyses']['vulnerability']
+        for vulnerability in vulnerabilities:
+            assert "cvss" in vulnerability
+            assert "is_private" in vulnerability
+            assert "vendor_cve_ids" in vulnerability
+
+
+@when('I try to access snyk link i should get {status:d} status code')
+def check_snyk_link_is_valid(context, status):
+    """Check if snyk link works."""
+    json_data = context.response.json()
+    url = json_data['registration_link']
+    response = requests.get(url)
+    assert response.status_code == status, "Unable to access the snyk login url"
+
+
+@then('I should find essential fields present in the result')
+def look_for_other_attributes(context):
+    """Check for all attributes."""
+    json_data = context.response.json()
+    assert "recommended_versions" in json_data, "No recommended version found"
+    assert "registration_link" in json_data, "No snyk registration link found"
+    assert "component_analyses" in json_data, "No component analyses data found"
+    assert "message" in json_data, "No message found"
+    assert "severity" in json_data, "No severity found"
+    assert "known_security_vulnerability_count" in json_data
+    assert "security_advisory_count" in json_data
+
+
+@then('I should get no result or recommendation with report')
+def check_for_recommendation_result_report(context):
+    """Check for result if recommedation or not."""
+    json_data = context.response.json()
+    if "recommendation" in json_data:
+        check_recommendation_in_result(context)
+    else:
+        look_for_other_attributes(context)
+        check_vulnerability_in_result(context)
+
+
+@then('I should find CVE report {cve} with score {score} in the component v2 analysis')
+def check_report_cve_score(context, cve, score):
+    """Check the cve report for specific cve id and score."""
+    json_data = context.response.json()
+
+    if "component_analyses" in json_data:
+        vulnerabilities = json_data['component_analyses']['vulnerability']
+        for v in vulnerabilities:
+            assert "cvss" in v
+            assert "is_private" in v
+            assert "vendor_cve_ids" in v
+            if v["vendor_cve_ids"] == cve and str(v["cvss"]) == score:
+                return
+    raise Exception("Can not find CVE {} with score {}".format(cve, score))
+
+
+@then('I should find a private vulnerability in v2 component analysis')
+def check_for_private_vul(context):
+    """Check for a private vulnerability in report."""
+    json_data = context.response.json()
+
+    if "component_analyses" in json_data:
+        vulnerabilities = json_data['component_analyses']['vulnerability']
+        for v in vulnerabilities:
+            assert "cvss" in v
+            assert "is_private" in v
+            assert "vendor_cve_ids" in v
```

Comment:
private vulnerability might not have cve_ids. I dont get this test. Please explain

Reply:
i tried for many it gives you CVE ids for example 
`{
    "recommended_versions": "2.3.9",
    "registration_link": "https://app.snyk.io/login",
    "component_analyses": {
        "vulnerability": [
            {
                "vendor_cve_ids": "SNYK-PYTHON-MARKDOWN2-40770",
                "cvss": "6.1",
                "is_private": false
            },
           ###  {
                "vendor_cve_ids": "SNYK-PYTHON-MARKDOWN2-568905",
                "cvss": "7.3",
                "is_private": true
            },
            {
                "vendor_cve_ids": "SNYK-PYTHON-MARKDOWN2-567097",
                "cvss": "7.1",
                "is_private": false
            }
        ]
    },
    "message": "markdown2 - 2.2.0 has 2 known security vulnerability and 1 security advisory with high severity. Recommendation: use version 2.3.9.",
    "severity": "high",
    "known_security_vulnerability_count": 2,
    "security_advisory_count": 1
}`

Reply:
you are looking at a wrong place. Look into the snyk feed. This is the data coming from our backbone. what if its wrong? Single pvt vulnerability will not have a cve (most of the time). This list you might be getting because there might be more than 1 vuln associated to pkg